File: EntityFrameworkCoreLibrary_vb/Contexts/NorthWindContext.vb
Imports System
Imports Microsoft.EntityFrameworkCore
Imports Microsoft.EntityFrameworkCore.Metadata
Imports EntityFrameworkCoreLibrary.Models
Imports System.Configuration.ConfigurationManager

Namespace EntityFrameworkCoreLibrary.Contexts
	Partial Public Class NorthWindContext
		Inherits DbContext

		Public Sub New()
		End Sub

        Public Sub New(options As DbContextOptions(Of NorthWindContext))
            MyBase.New(options)
        End Sub

        Public Overridable Property ContactType() As DbSet(Of ContactType)
		Public Overridable Property Contacts() As DbSet(Of Contacts)

        Protected Overrides Sub OnConfiguring(optionsBuilder As DbContextOptionsBuilder)
            If Not optionsBuilder.IsConfigured Then
                Dim environment = ""

#If Dev Then
                environment = "DevConnection"
#ElseIf Staging Then
				environment = "StagingConnection"
#Else
				environment = "ProductionConnection"
#End If
                optionsBuilder.UseSqlServer(AppSettings(environment))

            End If
        End Sub

        Protected Overrides Sub OnModelCreating(modelBuilder As ModelBuilder)

            modelBuilder.ApplyConfiguration(New ContactTypeConfiguration())
            modelBuilder.ApplyConfiguration(New ContactsConfiguration())

            OnModelCreatingPartial(modelBuilder)

        End Sub

        Partial Private Sub OnModelCreatingPartial(modelBuilder As ModelBuilder)
        End Sub
    End Class
End Namespace

File: EntityFrameworkCoreLibrary/Contexts/NorthWindContext.cs
using Microsoft.EntityFrameworkCore;
using EntityFrameworkCoreLibrary.Models;
using static System.Configuration.ConfigurationManager;

namespace EntityFrameworkCoreLibrary.Contexts
{
    public partial class NorthWindContext : DbContext
    {
        public NorthWindContext()
        {
        }

        public NorthWindContext(DbContextOptions<NorthWindContext> options)
            : base(options)
        {
        }

        public virtual DbSet<ContactType> ContactType { get; set; }
        public virtual DbSet<Contacts> Contacts { get; set; }

        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
            if (!optionsBuilder.IsConfigured)
            {
                var environment = "";

#if Dev
                environment = "DevConnection";
#elif Staging
                environment = "StagingConnection";
#else
                environment = "ProductionConnection";
#endif
                optionsBuilder.UseSqlServer(AppSettings[environment]);

            }
        }

        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            modelBuilder.ApplyConfiguration(new ContactTypeConfiguration());
            modelBuilder.ApplyConfiguration(new ContactsConfiguration());

            OnModelCreatingPartial(modelBuilder);
        }

        partial void OnModelCreatingPartial(ModelBuilder modelBuilder);
    }
}
